index.bs

```diff
@@ -812,18 +851,24 @@ spec:css-syntax-3;
         5.  Let |choice| be the result of <a abstract-op lt="ask to choose">asking the user to
             choose a `Credential`</a>, given |options| and |credentials|.
 
-        6.  If |choice| is `null` or a {{Credential}}, [=resolve=] |p| with |choice| and skip the
-            remaining steps.
+        6.  If |choice| is `null` or a {{Credential}}, let |result| be |choice| and go to step 9.
 
         7.  Assert: |choice| is an [=interface object=].
 
         8.  Let |result| be the result of executing |choice|'s
             {{[[DiscoverFromExternalSource]](origin, options, sameOriginWithAncestors)}},
             given |origin|, |options|, and |sameOriginWithAncestors|.
 
-        9.  If |result| is a {{Credential}} or `null`, resolve |p| with |result|.
+        9.  If |result| is a {{Credential}}, [=resolve=] |p| with |result|.
+
+        10. If |result| is an [=exception=], [=reject=] |p| with |result|.
```

Comment:
If this is possible, the old text failed to handle it... Is it possible?

Reply:
> If this is possible

Yes, [[DiscoverFromExternalSource]] can return an exception, see e.g. [step 4 on WebAuthn](https://w3c.github.io/webauthn/#sctn-discover-from-external-source).

> the old text failed to handle it

It did handle it, since an exception is not a Credential nor `null`, |p| would be rejected with |result|. This text keeps that behaviour (and makes it clearer I guess :P)